test/dynamics/solvers/test_dyson_magnus_solvers.py

```diff
@@ -74,6 +73,9 @@ def test_invalid_chebyshev_orders(self, SolverClass):
 class Test_PerturbativeSolver(QiskitDynamicsTestCase):
     """Tests for perturbative solver."""
 
+    def setUp(self):
+        self.jax_control_flow = False
+
```

Comment:
This is a bit of a weird case. It is technically possible here to use `test_array_backends` as you describe, and the extra tests included in `Test_PerturbativeSolverJAX` could be added as a totally separate test class.

The reason I didn't do this here though is that the method `Test_PerturbativeSolver.build_testing_objects`, which is run during `setUpClass` in both classes, is very expensive to execute. So, if I used `test_array_backends` to include JAX in `Test_PerturbativeSolver`, then created a separate test class for the extra JAX-transformation tests in `Test_PerturbativeSolverJAX`, the JAX version of executing `Test_PerturbativeSolver.build_testing_objects` would need to be run twice. I did it this way here to avoid that.

An alternate option could be to use `test_array_backends` with `Test_PerturbativeSolver`, and move the extra tests in `Test_PerturbativeSolverJAX` into that class, and have them get skipped if the backend isn't `"jax"`.

I think both the implemented approach, as well as the above alternate option, both suffer from inelegances. We could make this change but I'm also partial to keeping it as is.

Reply:
Thanks. I am for your original idea and can leave it as is.